types/react-transition-group/TransitionGroup.d.ts

```diff
@@ -1,19 +1,10 @@
-import { Component, HTMLAttributes, ReactElement, ReactType } from "react";
-import { HTMLTransitionGroupProps } from "./index";
+import { Component, ReactType, HTMLProps } from "react";
+import { TransitionActions } from "react-transition-group";
 
-declare namespace TransitionGroup {
-    interface TransitionGroupChildLifecycle {
-        componentWillAppear?(callback: () => void): void;
-        componentDidAppear?(): void;
-        componentWillEnter?(callback: () => void): void;
-        componentDidEnter?(): void;
-        componentWillLeave?(callback: () => void): void;
-        componentDidLeave?(): void;
-    }
-
-    type TransitionGroupProps = HTMLTransitionGroupProps<TransitionGroup>;
+export interface TransitionGroupProps extends TransitionActions, HTMLProps<any> {
+    component?: ReactType;
```

Comment:
This is a hard type to specify, as the `TransitionGroupProps` must also extend the type of the props of `component` (and not `HTMLProps<any>`).

I'm not sure if it _actually_ is possible to specify this correctly in the current typescript, but I took a stab here.

```ts
interface IntrinsicTransitionGroupProps<T extends keyof JSX.IntrinsicElements = 'div'> extends TransitionActions {
  component?: T
}
interface ComponentTransitionGroupProps<T extends ReactType> extends TransitionActions {
  component: T
}
type TransitionGroupProps<T extends keyof JSX.IntrinsicElements = 'div', U = any, V extends ReactType<U> = any> =
  (IntrinsicTransitionGroupProps<T> & JSX.IntrinsicElements[T]) | (ComponentTransitionGroupProps<V> & U)
```